src/diagnostics/riscv64/disasm-riscv64.cc

```diff
@@ -1537,33 +1537,45 @@ void Decoder::DecodeCRType(Instruction* instr) {
 }
 
 void Decoder::DecodeCAType(Instruction* instr) {
-  switch (instr->RvcFunct6Value()) {
-    case 0b100011:
-      switch (instr->RvcFunctValue()) {
-       case 0b00: //c.sub
-         if (instr->RvcRdValue() != 0 && instr->RvcRs2Value() != 0)
-           Format(instr, "sub       'Crs1s, 'Crs1s, 'Crs2s");
-         else
-           UNSUPPORTED_RISCV();
-         break;
-       default:
-         UNSUPPORTED_RISCV();
-      }
+  switch ((C1 | (instr->InstructionBits() & kCATypeMask) |
+          (instr->InstructionBits() & kRvcFunct2Mask))) {
```

Comment:
This condition could be simplified. Change `kCATypeMask` ->

```
const int kCATypeMask = kRvcOpcodeMask | kRvcFunct6Mask | kRvcFunct2Mask;
```

and then the switch could become

```
switch (instr->InstructionBits() & kCATypeMask) {
```